### PR TITLE
Add option to append VM UUID to generated friendly hostnames

### PR DESCRIPTION
**Pull request by zippolyte**
### What does this PR do?
SSIA

### Review checklist (to be filled by reviewers)

- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: tile/tile.yml [hunk 1]
```diff
@@ -211,22 +211,28 @@ forms:
...
   - name: use_friendly_hostname
     type: boolean
     label: Use Friendly Hostname
     default: true
     description: |
-      By default we use a friendly hostname, this might cause problems with some setups.
-      If you set the UUID hostname to true, it will supercede this one.
-      If you set both to false, the agent will detect the underlying hostname and use that instead.
+      Generate a friendly hostname for the host, in the form `<INSTANCE_NAME>-<INSTANCE_INDEX>`.
+      Check one or both of the options below to generate unique hostnames if you have several foundations with same deployments and/or several deployments with same VM instances.
```
Comment (nmuesch) [inline tile/tile.yml:221]: Can we use the option names directly instead of "below" to be explicit

File: tile/tile.yml [hunk 2]
```diff
@@ -211,22 +211,28 @@ forms:
     label: Use UUID Hostname
     default: false
     description: |
-      By default we use a friendly hostname, this might cause problems with some setups. Set this to use the UUID instead.
-      In an environment where there are multiple hosts of the same type on different deployments or in different foundries, this will be necessary.
-      If dd.hostname is set, it will take precedence over the uuid hostname.
```
Comment (nmuesch) [inline tile/tile.yml:216]: Is this still true, if so any reason to remove it?

**Review by nmuesch — COMMENTED** [review]
Minor nits on descriptions, but LGTM.

File: tile/tile.yml (hunk 2 above)
Comment (zippolyte) [reply tile/tile.yml:216]: Not true anymore, cause you can use the append_uuid_to_hostname option instead

File: tile/tile.yml [hunk 3]
```diff
@@ -218,14 +218,14 @@ forms:
     default: true
     description: |
       Generate a friendly hostname for the host, in the form `<INSTANCE_NAME>-<INSTANCE_INDEX>`.
-      Check one or both of the options below to generate unique hostnames if you have several foundations with same deployments and/or several deployments with same VM instances.
+      Check one or both of the options below (`Append VM UUID to friendly hostname` or `Make Friendly Hostname unique`) to generate unique hostnames if you have several foundations with same deployments and/or several deployments with same VM instances.
```
Comment (nmuesch) [inline tile/tile.yml:221]:
```suggestion
      Check one or both of the options below (`Append VM UUID to friendly hostname` or `Make Friendly Hostname unique`) to generate unique hostnames if you have several foundations with the same deployments and/or several deployments with the same VM instances.
```

**Review by nmuesch — APPROVED** [review]
Approved with a tiny doc suggestion.

**Pull request merged**